src/tests/ftest/cart/selftest/cart_selftest_three_node.yaml

```diff
@@ -30,7 +30,7 @@ tests: !mux
   self_np:
```

Comment:
Need an answer to ending question.  How are you going to handle file addresses when they are part of the input parameters and not and executable?  Also, input file also contains file names/paths.

Reply:
`avocado` will find executables in standard `PATH` locations:

- `/bin/`
- `/usr/bin/`
- `/opt/bin/`
- `/usr/local/bin/`
- `$HOME/bin/`

For `DAOS_ENV=dev`, executables can also be in this directory:

`$topdir/install/lib/daos/TESTING/tests`

For unset `DAOS_ENV` (CI/Jenkins), executables can also be in this directory

`/usr/lib/daos/TESTING/tests`

Our YAML parsing routines _**do not know**_ whether the client/server executables modify `PATH` env var, e.g., to find binaries passed to it via the command-line.

Reply:
You are missing the point.  I am not talking about executables but input/output files to self_test.  They are specified  as input parameters and as files within the input file.  Currently, they are specified as relative paths.
Self_test now has a function that can strip off the path proceeding a file name and replace it with an environmental variable.  But that feature  is not specified in the cart-946 implementation.  To do so, I would need the definition of environmental to use.

Reply:
> This code removes relative paths, so do you need the executable in the directory that that you are executing from or do you have a "PATH" environment which includes the executables?

The latter -- this PR changes `PATH` to include the paths which contain executables.

> How does this fit in with the modifications to self_test? Here we have the path to the selftest.cfg as a parameter option and not as an executable. Also, within selftest.cfg, there are relative paths to the expected and result file.

I expect the relative paths in the self_test input/output files to work as they have been working, since the current working directory is not changed in this PR.

Reply:
You are correct, the PR explicitly specifies relative paths to executables and not relative paths in general.  There is still an issue with relative paths that are not executable but you have resolved this PR.